docs/okeps/okep-5259-no-overlay.md

```diff
@@ -0,0 +1,598 @@
+
+For the cluster default network, the cluster admin shall create a
+ClusterUserDefinedNetwork CR with the reserved name `default` and set
+encapsulation to `None` to enable no-overlay mode before deploying
+OVN-Kubernetes. For this CR, OVN-K shall not create the corresponding
+NetworkAttachementDefinition CR. The subnet field must be aligned with the value
+in `cluster-subnets` parameter. The MTU shall not be larger than the node
+interface MTU.
+
+```yaml
+apiVersion: k8s.ovn.org/v1
+kind: ClusterUserDefinedNetwork
+Metadata:
+  # 'default' is the reserved name to represent the cluster default network
+  name: default
+spec:
+  # use an empty selector to select all namespaces
+  namespaceSelector: {}
+  network:
+    topology: Layer3
+    layer3:
+      role: Primary
+      mtu: 1500
+      subnets:
+      - cidr: 10.244.0.0/16
+        hostSubnet: 24
+      encapsulation: None
+```
+
+### Implementation Details
+
+No-overlay mode is only available when OVNK interconnect is enabled, since it
+relies on the route advertisement feature, which only works with interconnect
+mode.
+
+A network that is configured to use no-overlay mode needs a slightly modified
+OVN topology. It is useful to remember that in today's interconnect architecture
+a network is able to span all cluster nodes thanks to the cross-node logical
+links that OVN establishes between the transit switch and the ovn-cluster-router
+instances on each node. That is the logical representation of the OVN network
+and in practice these links are implemented through Geneve tunnels between every
+node in the cluster. For a network to skip the encapsulation step and route pod
+traffic directly over the underlay network, we need to change the logical
+network topology of this network: its ovn-cluster-router won't connect to the
+transit switch anymore and all east-west traffic will be naturally routed in the
+same path as north-south traffic through the provider network.
+
+In Shared Gateway Mode, at the OVN ClusterRouter, the pod-to-remote_pod egress
+traffic will be forwarded to the GatewayRouter via the join switch. Unlike the
+current route advertisement behavior, the routes to remote node subnets will be
+imported to the Gateway Router. Once the traffic reaches the GatewayRouter, it
+will be forwarded the nexthop node according to routes learnt from BGP.
+
+In Local Gateway Mode, the pod-to-remote_pod traffic egress traffic will enter
+the host VRF via the ovn-k8s-mpX interface. Then it will be forwarded to remote
+nodes according to the host routes of the VRF.
+
+The following diagram shows the OVN topology of a cluster with three networks:
+
+* a CUDN (in green) that uses Geneve encapsulation (default behavior)
+* a default network (in gray) that enables no-overlay mode
+* a CUDN (in blue) that enables no-overlay mode
+
+![No-overlay topology](../images/no_overlay_topology.jpg)
+
+#### No SNAT in no-overlay mode for egress traffic
```

Comment:
It is previously mentioned that SNAT will be kept in place for non-BGP next hops, but I am missing implementation details about this.